pkg/services/folder/folderimpl/folder.go

```diff
@@ -207,81 +215,86 @@ func (s *Service) CreateFolder(ctx context.Context, user *user.SignedInUser, org
+	query := models.GetDashboardQuery{OrgId: user.OrgID, Uid: cmd.Folder.UID}
 	if _, err := s.dashboardStore.GetDashboard(ctx, &query); err != nil {
-		return toFolderError(err)
+		return nil, toFolderError(err)
 	}
 
 	dashFolder := query.Result
 	currentTitle := dashFolder.Title
 
 	if !dashFolder.IsFolder {
-		return dashboards.ErrFolderNotFound
+		return nil, dashboards.ErrFolderNotFound
 	}
 
-	cmd.UpdateDashboardModel(dashFolder, orgID, user.UserID)
+	cmd.UpdateDashboardModel(dashFolder, user.OrgID, user.UserID)
 
 	dto := &dashboards.SaveDashboardDTO{
 		Dashboard: dashFolder,
-		OrgId:     orgID,
+		OrgId:     user.OrgID,
 		User:      user,
 		Overwrite: cmd.Overwrite,
 	}
 
 	saveDashboardCmd, err := s.dashboardService.BuildSaveDashboardCommand(ctx, dto, false, false)
 	if err != nil {
-		return toFolderError(err)
+		return nil, toFolderError(err)
 	}
 
 	dash, err := s.dashboardStore.SaveDashboard(ctx, *saveDashboardCmd)
 	if err != nil {
-		return toFolderError(err)
+		return nil, toFolderError(err)
 	}
 
-	var folder *models.Folder
-	folder, err = s.dashboardStore.GetFolderByID(ctx, orgID, dash.Id)
+	var modelFolder *models.Folder
+	modelFolder, err = s.dashboardStore.GetFolderByID(ctx, user.OrgID, dash.Id)
 	if err != nil {
-		return err
+		return nil, err
 	}
-	cmd.Result = folder
+	folder := f.ConvertModelFolderToFolder(modelFolder)
+	cmd.Folder = folder
 
 	if currentTitle != folder.Title {
 		if err := s.bus.Publish(ctx, &events.FolderTitleUpdated{
 			Timestamp: folder.Updated,
 			Title:     folder.Title,
 			ID:        dash.Id,
 			UID:       dash.Uid,
-			OrgID:     orgID,
+			OrgID:     user.OrgID,
 		}); err != nil {
 			s.log.Error("failed to publish FolderTitleUpdated event", "folder", folder.Title, "user", user.UserID, "error", err)
 		}
 	}
 
-	return nil
+	return cmd.Folder, nil
 }
 
-func (s *Service) DeleteFolder(ctx context.Context, user *user.SignedInUser, orgID int64, uid string, forceDeleteRules bool) (*models.Folder, error) {
-	dashFolder, err := s.dashboardStore.GetFolderByUID(ctx, orgID, uid)
+func (s *Service) DeleteFolder(ctx context.Context, cmd *folder.DeleteFolderCommand) (*folder.Folder, error) {
+	user, err := appcontext.User(ctx)
+	dashFolder, err := s.dashboardStore.GetFolderByUID(ctx, user.OrgID, string(user.UserID))
 	if err != nil {
 		return nil, err
 	}
 
-	guard := guardian.New(ctx, dashFolder.Id, orgID, user)
+	guard := guardian.New(ctx, dashFolder.Id, user.OrgID, user)
 	if canSave, err := guard.CanDelete(); err != nil || !canSave {
 		if err != nil {
 			return nil, toFolderError(err)
 		}
 		return nil, dashboards.ErrFolderAccessDenied
 	}
 
-	deleteCmd := models.DeleteDashboardCommand{OrgId: orgID, Id: dashFolder.Id, ForceDeleteFolderRules: forceDeleteRules}
+	folder := f.ConvertModelFolderToFolder(dashFolder)
+
+	deleteCmd := models.DeleteDashboardCommand{OrgId: user.OrgID, Id: dashFolder.Id, ForceDeleteFolderRules: forceDeleteRules} // how can i get forceDeleteRules from?
```

Comment:
how can i access `forceDeleteRules` here?

Reply:
this flag is [for deleting the alert rules under the folder or return error if there are any](https://github.com/grafana/grafana/pull/36427/files#diff-9b8062a98126348a456e2b1a6f09d54537cf6021c6b1e370f511f7adde4a2b71).
AFAICT deleting the folder contents during folder deletion is an open discussion but I would suggest for now adding an optional boolean flag `DeleteContent` on `folder.DeleteFolderCommand`
and set it in the API level [as it happens today](https://github.com/grafana/grafana/blob/92b5baba55e42c9898a1cbf38bd083b36063d221/pkg/api/folder.go#L99).